components/cosmetic_filters/renderer/cosmetic_filters_js_handler.cc

```diff
@@ -88,7 +88,8 @@ const char kHideSelectorsInjectScript[] =
           if (!document.adoptedStyleSheets.includes(
               window.content_cosmetic.cosmeticStyleSheet)) {
             document.adoptedStyleSheets =
-              [window.content_cosmetic.cosmeticStyleSheet];
+              [window.content_cosmetic.cosmeticStyleSheet,
+                ...document.adoptedStyleSheets];
```

Comment:
if this can overwrite a website's stylesheet, is it possible for websites to "clear" cosmetic filtering that we set this way too?

Reply:
Yes, it is

Reply:
perhaps a way to prevent that is to check and reapply when we do passes for 1st party to unhide